Assets/Scripts/Data Model/LayerManager.cs

```diff
@@ -1,948 +1,950 @@
+				{
+					List<EnergyLineStringSubEntity> newCablesToRemove = energyCableLayerGrey.RemoveInvalidCables();
+					if (newCablesToRemove != null && newCablesToRemove.Count > 0)
+						cablesToRemove.AddRange(newCablesToRemove);
+				}
+				energyCableLayerGrey.ActivateCableLayerConnections();
 			}
+			return cablesToRemove;
 		}
-	}
 
-	public static void UpdateVisibleLayersToBase()
-	{
-		foreach (AbstractLayer layer in visibleLayers)
+		public static void RestoreRemovedCables(List<EnergyLineStringSubEntity> removedCables)
 		{
-			//layer.SetEntitiesActiveUpToCurrentTime();
-			layer.SetEntitiesActiveUpTo(-1);
-			layer.RedrawGameObjects(CameraManager.Instance.gameCamera);
+			if (energyCableLayerGreen != null)
+			{
+				energyCableLayerGreen.RestoreInvalidCables(removedCables);
+			}
+			if (energyCableLayerGrey != null)
+			{
+				energyCableLayerGrey.RestoreInvalidCables(removedCables);
+			}
 		}
-		if (energyCableLayerGreen != null || energyCableLayerGrey != null)
+
+		/// <summary>
+		/// Sets entities in visible layers active to plan and shows layers in the plan that were not visible.
+		/// </summary>
+		/// <param name="plan"></param>
+		public static void UpdateVisibleLayersToPlan(Plan plan)
 		{
-			foreach (AbstractLayer energyLayer in energyLayers)
-				energyLayer.ResetCurrentGrids();
+			foreach (AbstractLayer layer in visibleLayers)
+			{
+				layer.SetEntitiesActiveUpTo(plan);
+				layer.RedrawGameObjects(CameraManager.Instance.gameCamera);
+			}
+
+			foreach (PlanLayer planLayer in plan.PlanLayers)
+				if (!visibleLayers.Contains(planLayer.BaseLayer))
+					ShowLayer(planLayer.BaseLayer);
+
+			if (energyCableLayerGreen != null || energyCableLayerGrey != null)
+			{
+				foreach (AbstractLayer energyLayer in energyLayers)
+					energyLayer.ResetCurrentGrids();
+
+				List<EnergyGrid> grids = PlanManager.GetEnergyGridsAtTime(plan.StartTime, EnergyGrid.GridColor.Either);
+				if (energyCableLayerGreen != null)
+				{
+					Dictionary<int, List<DirectionalConnection>> network = energyCableLayerGreen.GetCableNetworkForPlan(plan);
+					foreach (EnergyGrid grid in grids)
+						if (grid.IsGreen)
+							grid.SetAsCurrentGridForContent(network);
+				}
+				if (energyCableLayerGrey != null)
+				{
+					Dictionary<int, List<DirectionalConnection>> network = energyCableLayerGrey.GetCableNetworkForPlan(plan);
+					foreach (EnergyGrid grid in grids)
+						if (!grid.IsGreen)
+							grid.SetAsCurrentGridForContent(network);
+				}
+			}
 		}
-	}
 
-	public static void UpdateVisibleLayersToTime(int month)
-	{
-		foreach (AbstractLayer layer in visibleLayers)
+		public static void UpdateVisibleLayersToBase()
 		{
-			layer.SetEntitiesActiveUpToTime(month);
-			layer.RedrawGameObjects(CameraManager.Instance.gameCamera);
+			foreach (AbstractLayer layer in visibleLayers)
+			{
+				//layer.SetEntitiesActiveUpToCurrentTime();
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)